back-end/routes/recommendations.js

```diff
@@ -0,0 +1,90 @@
+const express = require('express');
+const { PrismaClient } = require('@prisma/client');
+const fetch = require('node-fetch');
+const router = express.Router();
+const prisma = new PrismaClient();
+
+router.use(express.json());
+
+const getAccessToken = async (userId) => {
+    const token = await prisma.token.findFirst({
+        where: {
+            userId: userId
+        },
+        orderBy: {
+            createdAt: 'desc'
+        }
+    });
+    return token ? token.token : null;
+};
+
+const fetchStitchAverages = async (songs, access_token) => {
+    const songIds = songs.map(song => (song.uri).split(':').pop());
+    const stringOfIds = songIds.join(',');
+    try {
+        const response = await fetch(`https://api.spotify.com/v1/audio-features?ids=${encodeURIComponent(stringOfIds)}`, {
+            headers: {
+                'Authorization': `Bearer ${access_token}`
+            }
+        });
+        const audioData = await response.json();
+    } catch (error) {
+        console.error(error);
+    }
+};
+
+const fetchRecommendations = async (songs, access_token) => {
+    const lastSongIds = songs.map(song => (song.uri).split(':').pop()).slice(-5);
+    const stringOfIds = lastSongIds.join(',');
+    try {
+        const response = await fetch(`https://api.spotify.com/v1/recommendations?limit=20&seed_tracks=${stringOfIds}`, {
+            headers: {
+                'Authorization': `Bearer ${access_token}`
+            }
+        });
+        const recommendData = await response.json();
+        const recommendations = recommendData.tracks;
+        const recommendIds = recommendations.map(track => (track.uri).split(':').pop());
+        return recommendIds;
+    } catch (error) {
+        console.error(error);
```

Comment:
No `console.error`s as discussed please, let's handle the errors instead.

https://www.freecodecamp.org/news/effective-error-handling-in-react-applications/#how-to-communicate-errors-to-users

Reply:
> No `console.error`s as discussed please, let's handle the errors instead.
> 
> https://www.freecodecamp.org/news/effective-error-handling-in-react-applications/#how-to-communicate-errors-to-users

Can we discuss this? I don't remember us discussing console.error's. Thank you

Reply:
As discussed on earlier PRs*